src/main/java/com/duckblade/osrs/toa/features/tomb/RewardSarcophagus.java

```diff
@@ -0,0 +1,277 @@
+public class RewardSarcophagus implements PluginLifecycleComponent
+{
+
+	private static final int[] VARBIT_MULTILOC_IDS_CHEST = new int[]{
+		14356, 14357, 14358, 14359, 14360, 14370, 14371, 14372
+	};
+
+	private static final int VARBIT_VALUE_CHEST_KEY = 2;
+	private static final int VARBIT_ID_SARCOPHAGUS = 14373;
+	private static final int WALL_OBJECT_ID_SARCOPHAGUS = 46221;
+
+	private final EventBus eventBus;
+	private final Client client;
+	private final TombsOfAmascutConfig config;
+	private final ClientThread clientThread;
+
+	private final List<WallObject> wallObjects = new ArrayList<>();
+
+	@Nullable
+	private int[] defaultFaceColors1;
+	@Nullable
+	private int[] defaultFaceColors2;
+	@Nullable
+	private int[] defaultFaceColors3;
+
+	private boolean sarcophagusIsPurple;
+	private boolean purpleIsMine = true;
+
+	@Override
+	public boolean isEnabled(final TombsOfAmascutConfig config, final RaidState raidState)
+	{
+		return raidState.getCurrentRoom() == RaidRoom.TOMB;
+	}
+
+	@Override
+	public void startUp()
+	{
+		eventBus.register(this);
+		parseVarbits();
+	}
+
+	@Override
+	public void shutDown()
+	{
+		eventBus.unregister(this);
+		resetFaceColors(wallObjects);
+		wallObjects.clear();
+		defaultFaceColors1 = null;
+		defaultFaceColors2 = null;
+		defaultFaceColors3 = null;
+		sarcophagusIsPurple = false;
+		purpleIsMine = true;
+	}
+
+	@Subscribe
+	public void onConfigChanged(final ConfigChanged event)
+	{
+		if (!event.getGroup().equals(TombsOfAmascutConfig.CONFIG_GROUP) || wallObjects.isEmpty()) return;
+
+		final String key = event.getKey();
+
+		if (key.startsWith("sarcophagus"))
+		{
+			clientThread.invokeLater(() -> recolor(wallObjects));
+		}
+	}
+
+	@Subscribe
+	public void onWallObjectSpawned(final WallObjectSpawned event)
+	{
+		final WallObject wallObject = event.getWallObject();
+
+		if (wallObject.getId() != WALL_OBJECT_ID_SARCOPHAGUS) return;
+
+		if (canCloneDefaultFaceColors())
+		{
+			final Model model = wallObject.getRenderable1().getModel();
+			defaultFaceColors1 = model.getFaceColors1().clone();
+			defaultFaceColors2 = model.getFaceColors2().clone();
+			defaultFaceColors3 = model.getFaceColors3().clone();
+		}
+
+		recolor(Collections.singleton(wallObject));
+
+		wallObjects.add(wallObject);
+	}
+
+	@Subscribe
+	public void onWallObjectDespawned(final WallObjectDespawned event)
+	{
+		final WallObject wallObject = event.getWallObject();
+
+		if (wallObject.getId() == WALL_OBJECT_ID_SARCOPHAGUS)
+		{
+			wallObjects.remove(wallObject);
+		}
+	}
+
+	private void parseVarbits()
+	{
+		clientThread.invokeLater(() ->
+		{
+			sarcophagusIsPurple = client.getVarbitValue(VARBIT_ID_SARCOPHAGUS) % 2 != 0;
+
+			for (final int varbitId : VARBIT_MULTILOC_IDS_CHEST)
+			{
+				if (client.getVarbitValue(varbitId) == VARBIT_VALUE_CHEST_KEY)
+				{
+					purpleIsMine = false;
+					break;
+				}
+			}
+
+			recolor(wallObjects);
+		});
+	}
+
+	private boolean canCloneDefaultFaceColors()
```

Comment:
If I understand correctly, this is just checking whether we already have the colours, which the method name doesn't indicate. `needDefaultFaceColours` or `missingDefaultFaceColours` would be more descriptive.